azurecompute-arm/src/main/java/org/jclouds/azurecompute/arm/compute/strategy/CreateResourcesThenCreateNodes.java

```diff
@@ -117,24 +114,22 @@ protected CreateResourceGroupThenCreateNodes(
 
       // This sill create the resource group if it does not exist
       String location = template.getLocation().getId();
-      ResourceGroup resourceGroup = resourceGroupMap.getUnchecked(location);
-      String azureGroupName = resourceGroup.name();
 
-      getOrCreateVirtualNetworkWithSubnet(location, options, azureGroupName);
-      configureSecurityGroupForOptions(group, azureGroupName, template.getLocation(), options);
+      createResourceGroupIfNeeded(group, location, options);
```

Comment:
I'm not really sure about this. If we do so, we should expose the method in the options that requires the object, which is not very convenient when it does not exist and you expect jclouds to create it for you. One possible option would be to keep the method to set the name, and add one to provide the ResourceGroup object, but that does not fix the side effect here: even if an object is set in the options, we need to check if that resource group exists and create it if missing, so the logic would be pretty much the same.
Regarding the jclouds code that consumes the resourceGroup option, the only thing that is used is actually the name, to create the different feature APIs, so the object is not really needed internally.
Thoughts?